### DISCUSSION TITLE
milvus-standalone always exited (2) on Macos 12.5.1

### DISCUSSION THREAD

**Discussion by htvinh** [milvus-io/milvus#18782] [Q&A and General discussion] [answered]
Hi all,
Milvus-standalone worked well when my MacOS was with 12.3.
When I upgrade MacOS to 12.5.1, I cannot install Milvus-standalone correctly.
As you see below, Milvus-standalone is always exited (2).

milvus-etcd         "etcd -advertise-cli…"   etcd                running              2379-2380/tcp
milvus-minio        "/usr/bin/docker-ent…"   minio               running (starting)   9000/tcp
milvus-standalone   "/tini -- milvus run…"   standalone          exited (2)    

And I found this message from the log in Docker:
panic: Corruption: Corruption: IO error: No such file or directoryWhile open a file for random read: /var/lib/milvus/rdb_data/000185.ldb: No such file or directory

Please help because I have used Milvus for a couple of projects until now.
Thanks,

**Comment by yhmo** [comment]
It cannot read file under the directory `/var/lib/milvus/rdb_data`, perhaps the path is not mapping correctly?
The standalone integrates a rocksdb as a data log pipeline, just like the Pulsar in cluster. All the insert/delete operations were persisted in rocksdb, in a directory which is configured in the milvus.yml:
```
rocksmq:
  path: /tmp/milvus/rdb_data # The path where the message is stored in rocksmq
```
If this directory is mapped to an outside(outside of the container) directory and mapped to an incorrect path, you may get this error.

To verify "standalone can work with MacOS 12.5.1", you can deploy a new milvus standalone container without any history data.

**Reply by htvinh** [reply]
Thanks for your answer. But I cannot understand how to deploy a new Milvus standalone container without any history data as you recommended ('without any history data' ???).
Please noted that I followed the Installation guide (https://milvus.io/docs/v2.1.x/install_standalone-docker.md).

**Reply by yhmo** [reply]
> Thanks for your answer. But I cannot understand how to deploy a new Milvus standalone container without any history data as you recommended ('without any history data' ???)

I mean create a new milvus standalone server on MacOS 12.5.1 by the two steps:
```
wget https://github.com/milvus-io/milvus/releases/download/v2.1.1/milvus-standalone-docker-compose.yml -O docker-compose.yml
sudo docker-compose up -d
```
If this new server works well, then we know there is no compatible problem.

**Reply by htvinh** [reply]
I follow your instruction but it doesn't work. 
When I do: 
sudo docker-compose ps
I got this (milvus-standalone started and existed)

NAME                COMMAND                  SERVICE             STATUS               PORTS
milvus-etcd         "etcd -advertise-cli…"   etcd                running              2379-2380/tcp
milvus-minio        "/usr/bin/docker-ent…"   minio               running (starting)   9000/tcp
milvus-standalone   "/tini -- milvus run…"   standalone          exited (2)

**Reply by yhmo** [reply]
Use this command to print the log:
`docker-compose logs -f`

**Reply by htvinh** [reply]
Here is the log:
[log.txt](https://github.com/milvus-io/milvus/files/9401324/log.txt)

milvus-minio       | WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
milvus-minio       |          Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
milvus-minio       | WARNING: Detected default credentials 'minioadmin:minioadmin', we recommend that you change these values with 'MINIO_ROOT_USER' and 'MINIO_ROOT_PASSWORD' environment variables
milvus-minio       | API: http://172.18.0.3:9000  http://127.0.0.1:9000 
milvus-minio       | 
milvus-minio       | Console: http://172.18.0.3:44269 http://127.0.0.1:44269 
milvus-minio       | 
milvus-minio       | Documentation: https://docs.min.io
milvus-minio       | 
milvus-minio       | WARNING: Console endpoint is listening on a dynamic port (44269), please use --console-address ":PORT" to choose a static port.
milvus-minio       | 
milvus-minio       |  You are running an older version of MinIO released 5 months ago 
milvus-minio       |  Update: Run `mc admin update` 
milvus-minio       | 
milvus-minio       | 
milvus-etcd        | {"level":"info","ts":1661246382.7705998,"caller":"flags/flag.go:113","msg":"recognized and used environment variable","variable-name":"ETCD_AUTO_COMPACTION_MODE","variable-value":"revision"}
milvus-etcd        | {"level":"info","ts":1661246382.770718,"caller":"flags/flag.go:113","msg":"recognized and used environment variable","variable-name":"ETCD_AUTO_COMPACTION_RETENTION","variable-value":"1000"}
milvus-etcd        | {"level":"info","ts":1661246382.7707658,"caller":"flags/flag.go:113","msg":"recognized and used environment variable","variable-name":"ETCD_QUOTA_BACKEND_BYTES","variable-value":"4294967296"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.770Z","caller":"etcdmain/etcd.go:72","msg":"Running: ","args":["etcd","-advertise-client-urls=http://127.0.0.1:2379","-listen-client-urls","http://0.0.0.0:2379","--data-dir","/etcd"]}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.773Z","caller":"etcdmain/etcd.go:115","msg":"server has been already initialized","data-dir":"/etcd","dir-type":"member"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.773Z","caller":"embed/etcd.go:131","msg":"configuring peer listeners","listen-peer-urls":["http://localhost:2380"]}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.774Z","caller":"embed/etcd.go:139","msg":"configuring client listeners","listen-client-urls":["http://0.0.0.0:2379"]}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.776Z","caller":"embed/etcd.go:307","msg":"starting an etcd server","etcd-version":"3.5.0","git-sha":"946a5a6f2","go-version":"go1.16.3","go-os":"linux","go-arch":"amd64","max-cpu-set":4,"max-cpu-available":4,"member-initialized":true,"name":"default","data-dir":"/etcd","wal-dir":"","wal-dir-dedicated":"","member-dir":"/etcd/member","force-new-cluster":false,"heartbeat-interval":"100ms","election-timeout":"1s","initial-election-tick-advance":true,"snapshot-count":100000,"snapshot-catchup-entries":5000,"initial-advertise-peer-urls":["http://localhost:2380"],"listen-peer-urls":["http://localhost:2380"],"advertise-client-urls":["http://127.0.0.1:2379"],"listen-client-urls":["http://0.0.0.0:2379"],"listen-metrics-urls":[],"cors":["*"],"host-whitelist":["*"],"initial-cluster":"","initial-cluster-state":"new","initial-cluster-token":"","quota-size-bytes":4294967296,"pre-vote":true,"initial-corrupt-check":false,"corrupt-check-time-interval":"0s","auto-compaction-mode":"revision","auto-compaction-retention":"1µs","auto-compaction-interval":"1µs","discovery-url":"","discovery-proxy":"","downgrade-check-interval":"5s"}
milvus-etcd        | {"level":"warn","ts":1661246382.7776818,"caller":"fileutil/fileutil.go:57","msg":"check file permission","error":"directory \"/etcd\" exist, but the permission is \"drwxr-xr-x\". The recommended permission is \"-rwx------\" to prevent possible unprivileged access to the data"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:42.797Z","caller":"etcdserver/backend.go:81","msg":"opened backend db","path":"/etcd/member/snap/db","took":"8.459096ms"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.291Z","caller":"etcdserver/server.go:508","msg":"recovered v2 store from snapshot","snapshot-index":300003,"snapshot-size":"8.2 kB"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.291Z","caller":"etcdserver/server.go:518","msg":"recovered v3 backend from snapshot","backend-size-bytes":294912,"backend-size":"295 kB","backend-size-in-use-bytes":192512,"backend-size-in-use":"192 kB"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.672Z","caller":"etcdserver/raft.go:483","msg":"restarting local member","cluster-id":"cdf818194e3a8c32","local-member-id":"8e9e05c52164694d","commit-index":340997}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d switched to configuration voters=(10276657743932975437)"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became follower at term 56"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"newRaft 8e9e05c52164694d [peers: [8e9e05c52164694d], term: 56, commit: 340997, applied: 300003, lastindex: 340997, lastterm: 56]"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","caller":"api/capability.go:75","msg":"enabled capabilities for version","cluster-version":"3.5"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","caller":"membership/cluster.go:276","msg":"recovered/added member from store","cluster-id":"cdf818194e3a8c32","local-member-id":"8e9e05c52164694d","recovered-remote-peer-id":"8e9e05c52164694d","recovered-remote-peer-urls":["http://localhost:2380"]}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.696Z","caller":"membership/cluster.go:285","msg":"set cluster version from store","cluster-version":"3.5"}
milvus-etcd        | {"level":"warn","ts":"2022-08-23T09:19:43.700Z","caller":"auth/store.go:1220","msg":"simple token is not cryptographically signed"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.701Z","caller":"mvcc/kvstore.go:345","msg":"restored last compact revision","meta-bucket-name":"meta","meta-bucket-name-key":"finishedCompactRev","restored-compact-revision":337853}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.702Z","caller":"mvcc/kvstore.go:415","msg":"kvstore restored","current-rev":338853}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.703Z","caller":"etcdserver/quota.go:117","msg":"enabled backend quota","quota-name":"v3-applier","quota-size-bytes":4294967296,"quota-size":"4.3 GB"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.703Z","caller":"etcdserver/server.go:834","msg":"starting etcd server","local-member-id":"8e9e05c52164694d","local-server-version":"3.5.0","cluster-id":"cdf818194e3a8c32","cluster-version":"3.5"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.703Z","caller":"etcdserver/server.go:728","msg":"started as single-node; fast-forwarding election ticks","local-member-id":"8e9e05c52164694d","forward-ticks":9,"forward-duration":"900ms","election-ticks":10,"election-timeout":"1s"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.705Z","caller":"embed/etcd.go:276","msg":"now serving peer/client/metrics","local-member-id":"8e9e05c52164694d","initial-advertise-peer-urls":["http://localhost:2380"],"listen-peer-urls":["http://localhost:2380"],"advertise-client-urls":["http://127.0.0.1:2379"],"listen-client-urls":["http://0.0.0.0:2379"],"listen-metrics-urls":[]}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.706Z","caller":"embed/etcd.go:580","msg":"serving peer traffic","address":"127.0.0.1:2380"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:43.706Z","caller":"embed/etcd.go:552","msg":"cmux::serve","address":"127.0.0.1:2380"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d is starting a new election at term 56"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became pre-candidate at term 56"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d received MsgPreVoteResp from 8e9e05c52164694d at term 56"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became candidate at term 57"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d received MsgVoteResp from 8e9e05c52164694d at term 57"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became leader at term 57"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.000Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"raft.node: 8e9e05c52164694d elected leader 8e9e05c52164694d at term 57"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.001Z","caller":"etcdserver/server.go:2027","msg":"published local member to cluster through raft","local-member-id":"8e9e05c52164694d","local-member-attributes":"{Name:default ClientURLs:[http://127.0.0.1:2379]}","request-path":"/0/members/8e9e05c52164694d/attributes","cluster-id":"cdf818194e3a8c32","publish-timeout":"7s"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.001Z","caller":"embed/serve.go:98","msg":"ready to serve client requests"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.002Z","caller":"etcdmain/main.go:47","msg":"notifying init daemon"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.002Z","caller":"etcdmain/main.go:53","msg":"successfully notified init daemon"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:19:44.002Z","caller":"embed/serve.go:140","msg":"serving client traffic insecurely; this is strongly discouraged!","address":"[::]:2379"}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:24:43.471Z","caller":"v3compactor/revision.go:86","msg":"starting auto revision compaction","revision":337853,"revision-compaction-retention":1000}
milvus-etcd        | {"level":"info","ts":"2022-08-23T09:24:43.471Z","caller":"v3compactor/revision.go:94","msg":"completed auto revision compaction","revision":337853,"revision-compaction-retention":1000,"took":"314.848µs"}
milvus-standalone  | 2022/08/23 09:19:43 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
milvus-standalone  | [2022/08/23 09:19:43.784 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]
milvus-standalone  | [2022/08/23 09:19:43.784 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]
milvus-standalone  | [2022/08/23 09:19:43.784 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]
milvus-standalone  | 
milvus-standalone  |     __  _________ _   ____  ______    
milvus-standalone  |    /  |/  /  _/ /| | / / / / / __/    
milvus-standalone  |   / /|_/ // // /_| |/ / /_/ /\ \    
milvus-standalone  |  /_/  /_/___/____/___/\____/___/     
milvus-standalone  | 
milvus-standalone  | Welcome to use Milvus!
milvus-standalone  | Version:   v2.1.1
milvus-standalone  | Built:     Fri Aug 12 11:05:57 UTC 2022
milvus-standalone  | GitCommit: bc62ca1f
milvus-standalone  | GoVersion: go version go1.16.9 linux/amd64
milvus-standalone  | 
milvus-standalone  | open pid file: /run/milvus/standalone.pid
milvus-standalone  | lock pid file: /run/milvus/standalone.pid
milvus-standalone  | [2022/08/23 09:19:43.785 +00:00] [INFO] [roles/roles.go:358] ["starting running Milvus components"]
milvus-standalone  | [2022/08/23 09:19:43.787 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]
milvus-standalone  | [2022/08/23 09:19:43.787 +00:00] [DEBUG] [paramtable/component_param.go:769] ["init cacheSize"] ["cacheSize (GB)"=0]
milvus-standalone  | [2022/08/23 09:19:43.789 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]
milvus-standalone  | [2022/08/23 09:19:43.789 +00:00] [DEBUG] [server/global_rmq.go:55] ["initializing global rmq"] [path=/var/lib/milvus/rdb_data]
milvus-standalone  | [2022/08/23 09:19:43.790 +00:00] [DEBUG] [server/global_rmq.go:98] [RocksmqRetentionTimeInMinutes=10080] [RocksmqRetentionSizeInMB=8192] [RocksmqPageSize=2147483648]
milvus-standalone  | [2022/08/23 09:19:43.790 +00:00] [DEBUG] [server/rocksmq_impl.go:171] ["Start rocksmq "] ["max proc"=4] [parallism=1] ["lru cache"=536870912]
milvus-standalone  | panic: Corruption: Corruption: IO error: No such file or directoryWhile open a file for random read: /var/lib/milvus/rdb_data/000185.ldb: No such file or directory
milvus-standalone  | 
milvus-standalone  | goroutine 1 [running]:
milvus-standalone  | github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc000152050, 0xc0001a2001, 0x0, 0x0)
milvus-standalone  | 	/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:373 +0x11b5
milvus-standalone  | github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc0001380c0, 0xc000196060, 0x3, 0x3, 0xc000112060)
milvus-standalone  | 	/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:111 +0x496
milvus-standalone  | github.com/milvus-io/milvus/cmd/milvus.RunMilvus(0xc000196060, 0x3, 0x3)
milvus-standalone  | 	/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x162
milvus-standalone  | main.main()
milvus-standalone  | 	/go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x45

**Reply by htvinh** [reply]
Here is the log in Docker:
2022/08/23 09:19:43 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined


    __  _________ _   ____  ______    

   /  |/  /  _/ /| | / / / / / __/    

  / /|_/ // // /_| |/ / /_/ /\ \    

 /_/  /_/___/____/___/\____/___/     


Welcome to use Milvus!

Version:   v2.1.1

Built:     Fri Aug 12 11:05:57 UTC 2022

GitCommit: bc62ca1f

GoVersion: go version go1.16.9 linux/amd64


open pid file: /run/milvus/standalone.pid

lock pid file: /run/milvus/standalone.pid

panic: Corruption: Corruption: IO error: No such file or directoryWhile open a file for random read: /var/lib/milvus/rdb_data/000185.ldb: No such file or directory


goroutine 1 [running]:

github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc000152050, 0xc0001a2001, 0x0, 0x0)

/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:373 +0x11b5

github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc0001380c0, 0xc000196060, 0x3, 0x3, 0xc000112060)

/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:111 +0x496

github.com/milvus-io/milvus/cmd/milvus.RunMilvus(0xc000196060, 0x3, 0x3)

/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x162

main.main()

/go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x45

[2022/08/23 09:19:43.784 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]

[2022/08/23 09:19:43.784 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]

[2022/08/23 09:19:43.784 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]

[2022/08/23 09:19:43.785 +00:00] [INFO] [roles/roles.go:358] ["starting running Milvus components"]

[2022/08/23 09:19:43.787 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]

[2022/08/23 09:19:43.787 +00:00] [DEBUG] [paramtable/component_param.go:769] ["init cacheSize"] ["cacheSize (GB)"=0]

[2022/08/23 09:19:43.789 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]

[2022/08/23 09:19:43.789 +00:00] [DEBUG] [server/global_rmq.go:55] ["initializing global rmq"] [path=/var/lib/milvus/rdb_data]

[2022/08/23 09:19:43.790 +00:00] [DEBUG] [server/global_rmq.go:98] [RocksmqRetentionTimeInMinutes=10080] [RocksmqRetentionSizeInMB=8192] [RocksmqPageSize=2147483648]

[2022/08/23 09:19:43.790 +00:00] [DEBUG] [server/rocksmq_impl.go:171] ["Start rocksmq "] ["max proc"=4] [parallism=1] ["lru cache"=536870912]

**Comment by yhmo** [comment, marked as answer]
Looks like the metadata and data are inconsistent. 
There is a folder named "volumes" under the same path of milvus-standalone-docker-compose.yml. Remove this folder and try `docker-compose up -d` again.

**Reply by htvinh** [reply]
I removed the folder named "volumes" and it works.
Many thanks @yhmo. You saved me.

**Reply by juanandreas** [reply]
Isn't that where the data is stored? Does removing this folder delete any preexisting collections/data?

**Reply by yhmo** [reply]
For standalone, all data is stored in the "volumes" folder. Removing this folder will lose all the preexisting collections.
Only when the data is cracked and milvus cannot restart, we remove this folder to reset the server.
